docs/jumpstart/week-5.md

```diff
@@ -0,0 +1,623 @@
+  // (you can access this locally at http://localhost:8081)
+  server.requestHandler(router).listen(defaultPort);
+
+  // Let the user know the server has started
+  System.out.println("Hey! The server has started on port " + defaultPort);
+}
+
+private void registerGetHiRoute(Router router) {
+  // Set the route up to respond to GET method requests.
+  Route hiRoute = router.get("/hi");
+  // Note: The hiRouteHandler method doesn't exist yet...
+  hiRoute.handler(this::hiRouteHandler);
+}
+
+private void hiRouteHandler(RoutingContext ctx) {
+  String resp = "Hey Jumpstarter!";
+  // Don't forget to set the response status!
+  end(ctx.response(), 200, resp);
+}
+
+private void handleHome(RoutingContext ctx) {
+...
+```
+
+#### Route Parameters
+
+What if you wanted to have a route parameter, that is, what if you wanted to have *dynamic* routes? Dynamic routes are
+routes that don't have to be explicitly declared. These can be in paths like `/jumpstart/hi`, `/<your_name>/hi`, or
+`/c4c/hi`, and then we can also have them return something like "hey <whatever you put in before 'hi'>". That can 
+actually be done pretty easily. Building off of the example above, this is what we would do.
+
+In the `registerGetHiRoute` method, we'll want to adjust the route to add the param.
+
+```java 
+private void registerGetHiRoute(Router router) {
+  // Set the route up to respond to GET method requests.
+  // Notice how we added a colon before the 'name' parameter.
+  Route hiRoute = router.get("/:name/hi");
+  // Note: The hiRouteHandler method doesn't exist yet...
+  hiRoute.handler(this::hiRouteHandler);
+}
+```
+
+And now that Vert.x knows to look for a parameter called 'name', it will be provided in the `RoutingContext`. We can now
+change `hiRouteHandler`. We've also provided the `getRequestParameterAsString` and `getRequestParameterAsInt` methods in 
+the `com.codeforcommunity.rest.RequestUtils` class to make getting that information a little easier (you'll probably 
+have to import it to `ApiMain`). Check it out if you're interested in how it works though!
+
+```java
+private void hiRouteHandler(RoutingContext ctx) {
+  // Request the 'name' param from the request
+  String name = getRequestParameterAsString(ctx.request(), "name");
+  String resp = "Hey " + name + "!";
+  // Don't forget to set the response status!
+  end(ctx.response(), 200, resp);
+}
+```
+
+So now the route should respond "Hey Jumpstarter!" when you navigate to `http://localhost:8081/Jumpstarter/hi`
+
+#### Query Parameters
+
+Query parameters are another kind of parameter that can be included. They are (sometimes optionally) included at the 
+end of the path and don't have to be included when creating the `Route` object from the `Router`. To be able to access 
+them, you can use `getRequestParameterAsString`. They're written in a path by starting with a `?` and chaining multiple
+together with `&`. Here are a some examples: `/some/path?param1=value1&param2=value2&param3=value3` and
```

Comment:
Put these examples on separate lines and include one that has more intuitive names that you would see in a real route